phoenix-core/src/it/java/org/apache/phoenix/end2end/UpdateCacheFreqIT.java

```diff
@@ -0,0 +1,230 @@
+            conn.commit();
+
+            final Statement statement = conn.createStatement();
+            ResultSet rs =
+                    statement.executeQuery(
+                            "SELECT COL2, VCOL1, VCOL2, COL5, COL6 FROM " + view02);
+            Assert.assertTrue(rs.next());
+            Assert.assertTrue(rs.next());
+            Assert.assertTrue(rs.next());
+            Assert.assertTrue(rs.next());
+            Assert.assertTrue(rs.next());
+            Assert.assertFalse(rs.next());
+
+            String query = "SELECT COL1, COL2, COL3, COL4 FROM " + tableName + " WHERE COL3 = " +
+                    "'col3_04'";
+            rs = statement.executeQuery(query);
+            Assert.assertTrue(rs.next());
+            Assert.assertFalse(rs.next());
+
+            ExplainPlan explainPlan = conn.prepareStatement(query)
+                    .unwrap(PhoenixPreparedStatement.class).optimizeQuery()
+                    .getExplainPlan();
+            ExplainPlanAttributes explainPlanAttributes =
+                    explainPlan.getPlanStepsAsAttributes();
+            Assert.assertEquals(indexName, explainPlanAttributes.getTableName());
+
+            TestUtil.removeCoprocessor(conn, "SYSTEM.CATALOG", TestMetaDataEndpointImpl.class);
+            TestUtil.addCoprocessor(conn, "SYSTEM.CATALOG", MetaDataEndpointImpl.class);
+        }
+    }
+
+    @Test
+    public void testUcfWithDisabledIndex() throws Throwable {
+        final String tableName = generateUniqueName();
+        final String indexName = "IDX_" + tableName;
+
+        Connection conn = DriverManager.getConnection(getUrl());
+        try {
+            final Statement stmt = conn.createStatement();
+
+            stmt.execute("CREATE TABLE " + tableName
+                    + " (COL1 CHAR(10) NOT NULL, COL2 CHAR(5) NOT NULL, COL3 VARCHAR,"
+                    + " COL4 VARCHAR CONSTRAINT pk PRIMARY KEY(COL1, COL2))"
+                    + " UPDATE_CACHE_FREQUENCY=10000");
+            stmt.execute("CREATE INDEX " + indexName
+                    + " ON " + tableName + " (COL3) INCLUDE (COL4)");
+
+            stmt.execute("ALTER INDEX " + indexName + " ON " + tableName + " DISABLE");
+
+            Thread.sleep(11000);
+            stmt.execute("UPSERT INTO " + tableName
+                    + " (col1, col2, col3, col4) values ('c001', "
+                    + "'c002', 'c003', 'c004')");
+            conn.commit();
+
+            TestUtil.removeCoprocessor(conn, "SYSTEM.CATALOG", MetaDataEndpointImpl.class);
+            TestUtil.addCoprocessor(conn, "SYSTEM.CATALOG", TestMetaDataEndpointImpl.class);
+
+            stmt.execute("UPSERT INTO " + tableName
+                    + " (col1, col2, col3, col4) values ('c011', "
+                    + "'c012', 'c013', 'c014')");
+            conn.commit();
+            throw new RuntimeException("Should not reach here");
+        } catch (PhoenixIOException e) {
```

Comment:
@virajjasani Do you think we should also test that once a "disabled" index is in BUILDING or ACTIVE state on the server, client learns about it and then stops overriding the UCF value?

Reply:
Except for Index jobs, do we any way to directly work on Index PTable object, bypassing data table PTable? If we rely on index jobs, it would be difficult to simulate when to switch over to new test coproc in the middle of job execution right?

Reply:
What if in the middle of this test:
1. we change index state using ALTER command
2. switch to normal co-proc and do a query on the table so that the getTable goes through
3. then we switch to new coproc and try another query to see if there was a getTable or not

Reply:
Query on the table will go through because the whole Data table PTable is in the client cache. We need query to Index table specifically.

Reply:
Done

Reply:
@virajjasani Maybe I am missing something but the current tests still only check whether there are no getTable calls when all indexes are active and that there are getTable calls when there is some index in a "disabled" state. 

Should there be a test where index is created in CREATE_DISABLE state to begin with, client should be overriding UCF and doing getTable calls, then we change the index state to BUILDING OR ACTIVE - after one more getTable call, client should stop overriding UCF and not do any getTable calls. Since this is exactly what would happen for atleast our prod use-case, I think it would be good to have that tested. WDYT?

Reply:
The reason why i am using ALTER TABLE to set index state is because otherwise using async index means running jobs to get the index state updated right? We want to avoid that and directly check index table state to begin with. The testing is same but avoiding async index status update is better as it tends to make tests flaky. We don't know by when the index state will be updated.